lib/util.js

```diff
@@ -672,22 +671,33 @@ function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   let visibleLength = 0;
   let index = 0;
-  while (index < value.length && visibleLength < ctx.maxArrayLength) {
-    let emptyItems = 0;
-    while (index < value.length && !hasOwnProperty(value, String(index))) {
-      emptyItems++;
-      index++;
-    }
-    if (emptyItems > 0) {
+  for (const elem in value) {
+    if (visibleLength === ctx.maxArrayLength)
+      break;
+    const i = +elem;
+    if (index !== i) {
+      if (i > 0 === false)
+        continue;
+      const emptyItems = i - index;
       const ending = emptyItems > 1 ? 's' : '';
       const message = `<${emptyItems} empty item${ending}>`;
       output.push(ctx.stylize(message, 'undefined'));
-    } else {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-                                 String(index), true));
-      index++;
+      index = i;
+      visibleLength++;
+      if (visibleLength === ctx.maxArrayLength)
```

Comment:
Couldn't these previous two lines be merged into `if (++visibleLength === ctx.maxArrayLength)` ?